### PR TITLE
Allow string property binding as Color source

### LINKED ISSUE #7683: TypeConverter not respected when evaluating Binding
### Description

When evaluating a `Binding` and the evaluated value type does not match the target type of the `BindableProperty`, the binding should check if a `TypeConverter` is available to perform the necessary conversion. This is convenient, as we do not always want to depend on UI-related types from `Microsoft.Maui.Controls`, `Microsoft.Maui.Graphics` etc. in the underlying `ViewModel`s.

This was working correctly in Xamarin.Forms, but I have discovered it is no longer working in MAUI, after migrating some code.

### Steps to Reproduce

Download and run the attached MAUI sample project.

[MauiTestApp.zip](https://github.com/dotnet/maui/files/8821746/MauiTestApp.zip)

The sample project has a single `Page` with a single `ViewModel`. The `ViewModel` exposes a collection of strings representing color names. The user can choose one of the color names with a `Picker` and that color name is bound to the `TextColor` property of two `Label`s. The first `Label` binds directly to the string property, assuming the `TypeConverter` of the `Color` type would kick-in to do the necessary conversion. The second `Label` binds to the string property via a `IValueConverter` like the following one:

```csharp
public class ValueTypeConverter : IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        var typeConverter = GetConverter(targetType);

        return typeConverter.ConvertFromInvariantString((string)value);
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        var typeConverter = GetConverter(targetType);

        return typeConverter.ConvertToInvariantString(value);
    }

    private static TypeConverter GetConverter(Type targetType)
    {
        // Code omitted for brevity.
    }
}
```

When selecting a color from the `Picker`, notice how the first `Label` does not change its `TextColor`, while the second one changes the `TextColor` correctly. Apparently the `TypeConverter` applied to the `Color` type is not respected.

I have double-checked and the `Color` type is declared like this in `Microsoft.Maui.Graphics`:

```csharp
namespace Microsoft.Maui.Graphics
{
    [DebuggerDisplay("Red={Red}, Green={Green}, Blue={Blue}, Alpha={Alpha}")]
    [TypeConverter(typeof(Converters.ColorTypeConverter))]
    public class Color
    {
        // Code omitted for brevity.
    }
}
```

For reference, I have attached a similar app in Xamarin.Forms:

[XamTestApp.zip](https://github.com/dotnet/maui/files/8821795/XamTestApp.zip)

Notice that in Xamarin.Forms both `Label`s change their `TextColor` accordingly.

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

N/A

### Did you find any workaround?

The workaround i have found is to implement a custom `IValueConverter` that simply calls the necessary `TypeConverter` to perform the desired conversion.

### Relevant log output

_No response_

### LINKED ISSUE #8390: Binding to color properties does not work
### Description

The color binding through Bindings does not work, I tried with the BoxView control with the Color property and the RoundRectangle control with the Fill property, both have the same problem.

### Steps to Reproduce

1.- Create a new project with the ".NET MAUI APP" template.
2.- Remove default content.
3.- In the XAML file, add StackLayout as the main layout:
```
    <StackLayout>
        <BoxView
        HeightRequest="200"
        WidthRequest="200"
        Color="{Binding .}" />
    </StackLayout>
```
4.- In the code behind, assign a color to the BindingContext:
```
	public MainPage()
	{
		InitializeComponent();
		BindingContext = "#000000";
	}
```
5.- Test the application

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Android 10, Windows 19044.1766

### Did you find any workaround?

Assigning the color with a value directly works (without the binding), however, in a CollectionView this is not feasible when you have multiple colors to be applied.

### Relevant log output

_No response_

### LINKED ISSUE #9662: Binding to hexColor doesn't work.
### Description

When I bind BackgroundColor of a button to hexColor in my ViewModel the app throws exception  `'Invalid cast from 'System.String' to 'Microsoft.Maui.Graphics.Color'.'` When I set BackgroundColor like this `<Button BackgroundColor="#fc87ad"/>` it works, but it doesnt work with a viewmodel.

### Steps to Reproduce

1. Create a new .NET MAUI project
2. Add button with BackgroundColor binding like this `<Button BackgroundColor="{Binding BackgroundButt}"/>`
3. Create ViewModel with `BackgroundButt` property:

 ```c#
    public class MyViewModel : INotifyPropertyChanged
    {
        public string BackgroundButt
        {
            get
            {
                return "#fc87ad";
            }
        }
       public event PropertyChangedEventHandler PropertyChanged;

       protected void OnPropertyChanged(string name)`
        {
            PropertyChangedEventHandler handler = PropertyChanged;
            if (handler != null)
            {
                handler(this, new PropertyChangedEventArgs(name));
            }
        }

    }
```

4. Add ViewModel to your page. The app throws exception `'Invalid cast from 'System.String' to 'Microsoft.Maui.Graphics.Color'.'
` 

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Allow string property binding as Color source.

image

### Issues Fixed

Fixes #9662
Fixes #8390
Fixes #7683

### CODE REVIEW

File: src/Controls/src/Core/BindableProperty.cs [hunk 1]
```diff
@@ -41,6 +41,7 @@ public sealed class BindableProperty
 		static readonly Dictionary<Type, TypeConverter> KnownTypeConverters = new Dictionary<Type, TypeConverter>
 		{
 			{ typeof(Uri), new UriTypeConverter() },
+			{ typeof(Maui.Graphics.Color), new ColorTypeConverter() },
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/BindableProperty.cs:44]: This is the main change.

**Comment by rmarinho** [comment]
can we add a xaml test?

**Review by hartez — DISMISSED** [review]

File: src/Controls/src/Core/BindableProperty.cs (hunk 1 above)
Comment (StephaneDelcroix) [reply src/Controls/src/Core/BindableProperty.cs:45]: That list is only for types we can’t attribute. Don’t we have a converter attribute on colour?

**Comment by jsuarezruiz** [comment]
> Doesn’t look like the correct way to fix this

Could you share more details?. Thanks!

**Comment by StephaneDelcroix** [comment]
> > Doesn’t look like the correct way to fix this
> 
> Could you share more details?. Thanks!

https://github.com/dotnet/maui/pull/9814#discussion_r1030645176

**Review by PureWeen — DISMISSED** [review]
https://github.com/dotnet/maui/pull/9814#issuecomment-1334984955

File: src/Controls/src/Core/BindableProperty.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/BindableProperty.cs:45]: Talked about this with Stephane and is ok to merge, Shane could you review it again?

**Review by rmarinho — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by StephaneDelcroix — APPROVED** [review]

**Pull request merged**